src/validation.h

```diff
@@ -881,6 +878,7 @@ class ChainstateManager
      */
     RecursiveMutex& GetMutex() const LOCK_RETURNED(::cs_main) { return ::cs_main; }
 
+    Options m_options;
```

Comment:
This will make internal code really verbose, forcing stuff like `m_options.m_max_tip_age` in every place an option is used (there are many places)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/25862#discussion_r951459817

> Maybe split up this change into another pull so that both pulls can work on the same base (maybe even avoiding a conflict) without risking to refactor the same lines of code twice in a short period of time?

Makes sense, now posted in #25905

Reply:
I wonder if the same should be done to blockman (#25781), but there the symbols are `chainstate.m_chainman.m_blockman.m_prune_mode`, which would turn into `chainstate.m_chainman.m_blockman.m_options.prune_mode`. Idk, both seem equally verbose.

Reply:
I'd probably opt to group the blockman options into a struct, but I don't think there is as much benefit because there are only 2 options instead of 6+.

I do see difference between the verbosity of `chainstate.m_chainman.m_blockman` (which is probably bad) and the verbosity of `m_blockman.m_options.prune_mode` (which is probably good). The verbosity in the first case is a "message chain" code smell (https://oowisdom.csse.canterbury.ac.nz/index.php/Message_chain_smell, https://en.wikipedia.org/wiki/Law_of_Demeter) indicating chainstate & chainman classes are might be doing too many things are and could be too closely coupled to block manager. The verbosity in the second case is caused by organizing related variables into a struct so they can be distinguished and treated as a logical unit.

I wouldn't want to hide an indication of poor code quality in one part of code by avoiding a potential improvement to a different part of the code. But I don't think stakes are very high, so any choice is probably reasonable here.